src/include/test/unit_matrix_with_ids.cc

```diff
@@ -0,0 +1,370 @@
+/**
+ * @file   unit_matrix_with_ids.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2024 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ */
+
+#include <algorithm>
+#include <catch2/catch_all.hpp>
+#include <vector>
+#include "cpos.h"
+#include "detail/linalg/matrix_with_ids.h"
+#include "mdspan/mdspan.hpp"
+
+TEST_CASE("matrix_with_ids: test test", "[matrix_with_ids]") {
+  REQUIRE(true);
+}
+
+TEMPLATE_TEST_CASE(
+    "matrix_with_ids: move constructor",
+    "[matrix_with_ids]",
+    stdx::layout_right,
+    stdx::layout_left) {
+  size_t rows = 5;
+  size_t cols = 10;
+  auto vectors = std::unique_ptr<float[]>(new float[rows * cols]);
+  auto ids = std::unique_ptr<size_t[]>(new size_t[rows * cols]);
+  auto matrix = MatrixWithIds<float, TestType>{
+      std::move(vectors), std::move(ids), rows, cols};
+}
+
+TEMPLATE_TEST_CASE(
+    "matrix_with_ids: size constructor",
+    "[matrix_with_ids]",
+    char,
+    float,
+    int32_t,
+    int64_t) {
+  auto matrix =
+      MatrixWithIds<TestType, stdx::layout_right, size_t, TestType>{2, 5};
+}
```

Comment:
Yep and we do check row and column orientations in several places with `TEMPLATE_TEST_CASE("..., "...", stdx::layout_right, stdx::layout_left) {`, but thanks, another unit test is always good, added this one.

Reply:
In general, `stdx::layout_right` is not well-tested at all.  We might want to audit that and make sure it works everywhere as well as `stdx::layout_left`.  The idea at one point was that a feature vector array should be agnostic as to which layout and that we would use a feature vector array everywhere rather than a `ColMajorMatrix`.  I have started putting concepts in many of the function templates as a start to that.

The layout doesn't really matter, because as long as the underlying one-dimensional data has each of the vectors as contiguous data, both row major and column major will have, respectively, rows and columns as contiguous data.  We don't ever iterate over a matrix element by element, but rather `operator[]` pulls out a row or column, i.e., a contiguous vector.  

For the same reason, I think the before and after calls in Python to `transpose` are superfluous.

In case you are wondering why I picked a column major matrix to represent a feature vector array -- in linear algebra a vector is almost universally represented as a column.